app/src/main/java/org/schabi/newpipe/player/helper/LockManager.java

```diff
@@ -18,10 +17,9 @@
     private WifiManager.WifiLock wifiLock;
 
     public LockManager(final Context context) {
-        powerManager = ((PowerManager) context.getApplicationContext()
-                .getSystemService(POWER_SERVICE));
-        wifiManager = ((WifiManager) context.getApplicationContext()
-                .getSystemService(WIFI_SERVICE));
+        powerManager = ContextCompat.getSystemService(context.getApplicationContext(),
```

Comment:
Why getApplicationContext()? Isn't context already enough, like the line below?

Reply:
The documentation for `WifiManager` states:

"On releases before Build.VERSION_CODES.N, this object should only be obtained from an Context#getApplicationContext(), and not from any other derived context to avoid memory leaks within the calling process."